tests/test_compatibility.gd

```diff
@@ -0,0 +1,114 @@
+extends GutTest
+## Tests for scene file backwards compatibility
+
+
+func get_scene_state_node_index(state: SceneState, name: String) -> int:
+	for idx in state.get_node_count():
+		if state.get_node_name(idx) == name:
+			return idx
+	return -1
+
+
+func get_scene_state_node_prop_index(state: SceneState, node_idx: int, name: String) -> int:
+	for prop_idx in state.get_node_property_count(node_idx):
+		if state.get_node_property_name(node_idx, prop_idx) == name:
+			return prop_idx
+	return -1
+
+
+func _get_block_names_recursive(block: Resource, names: Array[String]):
+	names.append(block.name)
+	if "children" in block:
+		for child in block.children:
+			_get_block_names_recursive(child, names)
+	for value in block.arguments.values():
+		if not value is Resource:
+			continue
+		if not value.script:
+			continue
+		if value.script.resource_path != "res://addons/block_code/serialization/value_block_serialization.gd":
+			continue
+		_get_block_names_recursive(value, names)
+
+
+func get_block_script_block_names(block_script: Resource) -> Array[String]:
+	var names: Array[String]
+	for tree in block_script.block_serialization_trees:
+		_get_block_names_recursive(tree.root, names)
+	return names
+
+
+func test_simple_spawner():
+	const old_block_names: Array[String] = [
+		"simplespawner_get_spawn_frequency",
+		"simplespawner_set_spawn_frequency",
+	]
+
+	const new_block_names: Array[String] = [
+		"simplespawner_get_spawn_period",
+		"simplespawner_set_spawn_period",
+	]
+
+	var scene: PackedScene = load("res://tests/data/simple_spawner_compat.tscn")
```

Comment:
Ooh, it's marvellous to have an example of a test with a scene file! Thank you for writing tests :)